### PR TITLE
feat: add pr_name config field

### LINKED ISSUE #1740: Add configuration field to customize release PR name
## Motivations

Not everyone likes the format `chore: release v0.2.31`.

## Solution

Add a [config](https://github.com/MarcoIeni/release-plz/blob/46f6489613eeb5a68d01c7845f0622f722c80d6a/website/docs/config.md#L52) field called `pr_name`. Similar to the `git_tag_name` field, the `pr_name` field should be a tera template.

The field `{{ version }}` should be available

## Additional context

Requested in https://github.com/MarcoIeni/release-plz/issues/1738

### PR DESCRIPTION
**Pull request by JustBeYou**
Close #1740

### CODE REVIEW

File: crates/release_plz_core/src/command/release_pr.rs [hunk 1]
```diff
@@ -415,6 +415,7 @@ async fn github_create_release_branch(
     release_branch: &str,
 ) -> anyhow::Result<()> {
     repository.push(release_branch)?;
+    // TODO @justbeyou would we like to create arguments for those hardcoded commit messages also?
```
Comment (MarcoIeni) [inline crates/release_plz_core/src/command/release_pr.rs:418]: Yes, I think it's what we need to do, right?

File: crates/release_plz_core/src/pr.rs [hunk 2]
```diff
@@ -64,20 +64,23 @@ fn pr_title(
             .all(|(_, update)| &update.version == first_version)
     };
 
+    // TODO @justbeyou seems like this piece of code is not covered by tests
+    // TODO @justbeyou we should probably add here the option to process a template if available
+    // otherwise we should keep the old behaviour
```
Comment (MarcoIeni) [inline crates/release_plz_core/src/pr.rs:69]: yes, exactly.

Regarding tests, there are some integration tests:
https://github.com/MarcoIeni/release-plz/blob/8d2bf351c29d6a7012444030441ca1d162af78b6/crates/release_plz/tests/all/main.rs#L1

File: crates/release_plz_core/src/command/release_pr.rs (hunk 1 above)
Comment (JustBeYou) [reply crates/release_plz_core/src/command/release_pr.rs:418]: I think so, but I observed that the commit message and PR title had different sources for their messages and I was wondering if this should also be changed in this PR.

Anyway, I was just exploring the codebase and writing some comments for myself to familiarize with the codebase.

Comment (MarcoIeni) [reply crates/release_plz_core/src/command/release_pr.rs:418]: I see :)

> I observed that the commit message and PR title had different sources for their messages and I was wondering if this should also be changed in this PR.

Oh, I see. Yeah, the PR name and the commit message should be the same. Feel free to do this refactoring in this pr or another one 👍

**Comment by JustBeYou** [comment]
@MarcoIeni, ready for review on this one 👀 

Short summary:
* did a little refactor of building Tera templates in `project.rs` and `tera.rs` as there was a little bit of annoying duplication
* implemented the PR name behavior and added tests
* synchronized the commit message in the release branch with the PR name

File: crates/release_plz_core/src/pr.rs [hunk 3]
```diff
@@ -64,7 +73,21 @@ fn pr_title(
...
+            &updates[0].0.name
+        } else {
+            ""
+        };
+        let first_version = first_version.to_string().clone();
+        let version = if are_all_versions_equal() {
+            first_version.as_str()
+        } else {
+            ""
+        };
```
Comment (MarcoIeni) [inline crates/release_plz_core/src/pr.rs:87]: sorry, I made a mistake, we shouldn't set the field if not present.
I.e. we shouldn't set empty string.

Comment (JustBeYou) [reply crates/release_plz_core/src/pr.rs:87]: should we let them unset? what would be the behavior in this case? the literal `{{ version }}` string appear in the message? or should we error?

Comment (JustBeYou) [reply crates/release_plz_core/src/pr.rs]: Failing seems the sanest way of handling things, as I think this feature will be used in very specific cases (looking at the original request). I added a commit to implement the change, let me know what you think

Comment (MarcoIeni) [reply crates/release_plz_core/src/pr.rs]: Imo we should not set the fields. But i haven't looked into your changes yet.

**Review by MarcoIeni — APPROVED** [review]
Incredible contribution. Very high quality of the code. Many improvements to the existing codebase. Thanks 🙌

I hope you enjoyed working on release-plz and you will contribute in the future too. If you want to chat with me about anything, feel free to DM in my socials 👍

**Comment by JustBeYou** [comment]
Thanks a lot for your support, it was a really awesome experience! I will take a look at some more issues in the following days and come back with some PRs.

**Pull request merged**